doc/stdlib/libc.rst

```diff
@@ -8,7 +8,7 @@
 
    .. Docstring generated from Julia source
 
-   Call ``malloc`` from the C standard library.
+   Call ``malloc`` from the C standard library. Note, this is not the standard way to allocate memory in Julia (there is no need to allocate or deallocate manually). This indirect way of "manual memory allocation", using the C Standard Library, will result in a memory leak if the memory is not freed later explicitly, e.g. it will not be handled by Julia's garbage collection mechanism. This use, is for Julia's internal standard library use or those who really know what they are doing and require more control. See also the embedding section of the manual before proceding, if your intention is really to call C functions that allocate memory.
```

Comment:
It might be nice to emphasize the need of manually freeing the memory although I personally feel like `C malloc` already cover that all.

I'm not sure what is the opposite of `not the standard way`. Also not sure what do you mean by `indirect way of`. I still don't think you should mention `for ... internal ... use` because there's nothing internal about it. This is also not related to embedding or the julia GC (which is described in the embedding section and I guess that you mean by `embedding section of the manual`). This is completely independent of the GC and you shouldn't need to understand the Julia GC in order to use `malloc`. It doesn't give you "more control" either.

To summary, this is allocating c memory and not GC memory so the difference is the type of memory, not how standard/direct/controllable they are. You use this if the c library requires it and that's pretty much it. I don't really think the text you've added explains that better than just ``call `malloc` in libc``.